### PR TITLE
REL-518430 | WorkspaceHelpers updated to contain both create and dele…

### PR DESCRIPTION
**Pull request by TommyBigNose**
…te functions with removed RSAPI code (general cleanup as well on the code)

### CODE REVIEW

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.Example.NUnit/TestTemplate.cs [hunk 1]
```diff
@@ -76,7 +73,7 @@ public void Execute_TestFixtureSetup()
 
 
 			//Create workspace
-			_workspaceId = WorkspaceHelpers.CreateWorkspace.CreateWorkspaceAsync(_workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, servicesManager, SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
+			_workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(helper.GetServicesManager(), _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.Example.NUnit/TestTemplate.cs:76]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/DocumentHelperIntegrationTests.cs [hunk 2]
```diff
@@ -30,9 +29,7 @@ public void SetUp()
 			testHelper = new TestHelper(configDictionary);
 			_workspaceName = $"IntTest_{Guid.NewGuid()}";
 			_servicesManager = testHelper.GetServicesManager();
-			_workspaceId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
+			_workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/DocumentHelperIntegrationTests.cs:32]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/FieldsHelperIntegrationTests.cs [hunk 3]
```diff
@@ -51,14 +43,14 @@ public void SetUp()
...
-			//Create workspace
-			_workspaceId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager, SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).ConfigureAwait(false).GetAwaiter().GetResult();
-
-			//Query for field ID to be used in test
-			_testFieldId = CreateTestField(_serviceFactory, _testFieldName, _workspaceId);
-			
+			_serviceFactory = GetServiceFactory();
+
+			//Create workspace
+			_workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/FieldsHelperIntegrationTests.cs:49]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/FoldersHelperIntegrationTests.cs [hunk 4]
```diff
@@ -36,9 +31,7 @@ public void SetUp()
 
 			_workspaceName = $"IntTest_{Guid.NewGuid()}";
 			_servicesManager = testHelper.GetServicesManager();
-			_workspaceId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
+			_workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/ArtifactHelpers/FoldersHelperIntegrationTests.cs:34]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/Kepler/ApplicationInstallHelperTests.cs [hunk 5]
```diff
@@ -46,7 +45,8 @@ public void SetUp()
 
 			Sut = new ApplicationInstallHelper(applicationInstallManager, libraryApplicationManager, ConfigurationHelper.SERVER_BINDING_TYPE, ConfigurationHelper.RELATIVITY_INSTANCE_ADDRESS, ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD);
 
-			_workspaceId = CreateWorkspace.Create(rsapiClient, "Test-" + Guid.NewGuid().ToString(), ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME);
+			string workspaceName = "Test-" + Guid.NewGuid().ToString();
+			_workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_testHelper.GetServicesManager(), workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/Kepler/ApplicationInstallHelperTests.cs:49]: Should use .ConfigureAwait(false)

**Review by calimetikcura — CHANGES_REQUESTED** [review]

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestHelperIntegrationTests.cs [hunk 6]
```diff
@@ -33,22 +27,16 @@ public void SetUp()
...
-
-			_workspaceOneId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
-
-			_workspaceTwoId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
+			_workspaceOneId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
+			_workspaceTwoId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestHelperIntegrationTests.cs:31]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestHelperIntegrationTests.cs [hunk 7]
```diff
@@ -33,22 +27,16 @@ public void SetUp()
...
 			_servicesManager = SuT.GetServicesManager();
-
-			_workspaceOneId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
-
-			_workspaceTwoId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
+			_workspaceOneId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(_servicesManager, _workspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestHelperIntegrationTests.cs:30]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestImportApiHelper.cs [hunk 8]
```diff
@@ -51,12 +46,12 @@ public void CreateDocumentsWithFolderName()
...
-			}
-
-			// Act
+			}
+
+			// Act
 			try
 			{
-				workspaceId = CreateWorkspace.CreateWorkspaceAsync(sampleWorkspaceName, ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, Sut.GetServicesManager(), ConfigurationHelper.ADMIN_USERNAME, ConfigurationHelper.DEFAULT_PASSWORD).Result;
+				workspaceId = Helpers.WorkspaceHelpers.WorkspaceHelpers.CreateAsync(Sut.GetServicesManager(), sampleWorkspaceName, SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME).Result;
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/TestImportApiHelper.cs:54]: Should use .ConfigureAwait(false)

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/WorkspaceHelpers/CreateWorkspaceHelperTests.cs [hunk 9]
```diff
@@ -37,29 +28,25 @@ public void SetUp()
...
 		}
 
 		[Test]
 		public void CreateWorkspaceTest()
-		{
-			_workspaceId = CreateWorkspace.CreateWorkspaceAsync(_workspaceName,
-				SharedTestHelpers.ConfigurationHelper.TEST_WORKSPACE_TEMPLATE_NAME, _servicesManager,
-				SharedTestHelpers.ConfigurationHelper.ADMIN_USERNAME, SharedTestHelpers.ConfigurationHelper.DEFAULT_PASSWORD).Result;
-
+		{
```
Comment (aarongilbert123) [inline Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/WorkspaceHelpers/CreateWorkspaceHelperTests.cs:38]: Should use .ConfigureAwait(false)

Comment (TommyBigNose) [reply Source/Relativity.Test.Helpers/Relativity.Test.Helpers.NUnit.Integration/WorkspaceHelpers/CreateWorkspaceHelperTests.cs:38]: I changed this but it doesn't appear to be showing up haha

**Review by aarongilbert123 — APPROVED** [review]

File: Source/Relativity.Test.Helpers/Relativity.Test.Helpers.Example.NUnit/packages.config [hunk 10]
```diff
@@ -17,5 +18,5 @@
   <package id="Relativity.Transfer.Client" version="6.0.16" targetFramework="net462" />
   <package id="RelativityDev.DbContextHelper" version="5.0.0.1" targetFramework="net462" />
```
[the remark this answers is not in the document]
Comment (TommyBigNose) [reply Source/Relativity.Test.Helpers/Relativity.Test.Helpers.Example.NUnit/packages.config:18]: I don't know if I can cut out this package just yet, TestHelpers is still using it in public IDBContext GetDBContext(int caseID).  I can look into this for a separate PR?

[the remark this answers is not in the document]
Comment (TommyBigNose) [reply Source/Relativity.Test.Helpers/Relativity.Test.Helpers.Example.NUnit/packages.config:18]: Out of scope right now, resolving

**Review by calimetikcura — APPROVED** [review]

**Pull request merged**